Fix some -Wfloat-conversion warnings (Pt. 10)

Holzhaus opened the pull request:
As reported here: https://github.com/mixxxdj/mixxx/pull/3192#discussion_r512705744
Compiler output: https://gist.github.com/poelzi/a1acac5d0a2109c5b6ed890cd422cd18

@poelzi Please test.

Holzhaus committed:
BroadcastManager: Fix implicit cast from double to bool

Fixes this warning:

    src/broadcast/broadcastmanager.cpp:97:31: warning: implicit conversion turns floating-point number into integer: 'double' to 'bool' [-Wfloat-conversion]
            emit broadcastEnabled(v);
                 ~~~~~~~~~~~~~~~~ ^
    1 warning generated.

Holzhaus committed:
Use ControlObject::toBool() method to fix -Wfloat-conversion errors

This fixes a bunch of warnigs regarding implicit casts from double
to bool.

Holzhaus committed:
Replace implicit casts from double to bool with comparisons

This fixes a lot of -Wfloat-conversion warnings on certain compilers.

Holzhaus committed:
Add missing typecasts to fix -Wfloat-comparison warnings

Holzhaus committed:
RateControl: Move constants into anonymous namespace

poelzi commented:
Still warnings:
https://gist.github.com/poelzi/4f55cb6e424919cc340cadb4b0d7d2cc

Holzhaus committed:
Replace implicit casts with ControlObject::toBool() in more places

Holzhaus committed:
Use more comparisons instead of implicit casts to double

Holzhaus committed:
effects: Fix cast of EffectManifestParameter::LinkType to int

Holzhaus committed:
vinylcontrolxwax: Add missing braces around if block

Holzhaus committed:
Fix wrong types of anonymous namespace constants

Holzhaus commented:
> Still warnings:
> https://gist.github.com/poelzi/4f55cb6e424919cc340cadb4b0d7d2cc

Thanks. Should be fixed, please test again.

uklotzde commented on src/control/controlobject.cpp:
```diff
@@ -94,6 +94,12 @@ double ControlObject::get(const ConfigKey& key) {
 }
 
+// static
+bool ControlObject::toBool(const ConfigKey& key) {
```
- Why not inline?
- Shrink to a single line without parentheses

Holzhaus replied:
Inline didn't work, it caused linker errors:

```
/usr/bin/ld: /usr/bin/ld: DWARF error: could not find variable specification at offset 84f37
/usr/bin/ld: DWARF error: could not find variable specification at offset 84f45
/usr/bin/ld: DWARF error: could not find variable specification at offset c3765
/usr/bin/ld: DWARF error: could not find variable specification at offset c396c
/usr/bin/ld: DWARF error: could not find variable specification at offset c39bd
libmixxx-lib.a(mixxx.cpp.o): in function `MixxxMainWindow::confirmExit()':
/home/jan/Projects/mixxx/src/mixxx.cpp:1607: undefined reference to `ControlObject::toBool(ConfigKey const&)'
/usr/bin/ld: /home/jan/Projects/mixxx/src/mixxx.cpp:1614: undefined reference to `ControlObject::toBool(ConfigKey const&)'
/usr/bin/ld: /usr/bin/ld: DWARF error: could not find variable specification at offset 84f37
/usr/bin/ld: DWARF error: could not find variable specification at offset 84f45
/usr/bin/ld: DWARF error: could not find variable specification at offset c3765
/usr/bin/ld: DWARF error: could not find variable specification at offset c396c
/usr/bin/ld: DWARF error: could not find variable specification at offset c39bd
libmixxx-lib.a(mixxx.cpp.o): in function `MixxxMainWindow::confirmExit()':
/home/jan/Projects/mixxx/src/mixxx.cpp:1607: undefined reference to `ControlObject::toBool(ConfigKey const&)'
/usr/bin/ld: /home/jan/Projects/mixxx/src/mixxx.cpp:1614: undefined reference to `ControlObject::toBool(ConfigKey const&)'
collect2: error: ld returned 1 exit status
```

Holzhaus replied:
The other changes are done in: https://github.com/mixxxdj/mixxx/pull/3227/commits/3981e1bf7d45489f772789b8316fe260c4fc20fa

Holzhaus replied:
Doh, I forgot. sorry. Fixed now.

uklotzde commented on src/effects/builtin/balanceeffect.cpp:
```diff
@@ -77,7 +77,7 @@ BalanceGroupState::BalanceGroupState(const mixxx::EngineParameters& bufferParame
           m_pHighBuf(MAX_BUFFER_LEN),
           m_oldSampleRate(bufferParameters.sampleRate()),
-          m_freq(kMinCornerHz),
+          m_freq(static_cast<int>(kMinCornerHz)),
```
m_freq should become a double

Holzhaus replied:
https://github.com/mixxxdj/mixxx/pull/3227/commits/5b2c009520e1ee3bdd5c29bb477dd09000f51ad0

uklotzde commented on src/vinylcontrol/vinylcontrolxwax.cpp:
```diff
@@ -256,7 +256,7 @@ void VinylControlXwax::analyzeSamples(CSAMPLE* pSamples, size_t nFrames) {
 
     // Check if vinyl control is enabled...
-    m_bIsEnabled = enabled == NULL ? false : checkEnabled(m_bIsEnabled, enabled->get());
+    m_bIsEnabled = (!enabled) ? false : checkEnabled(m_bIsEnabled, enabled->toBool());
```
`m_bIsEnabled = enabled && checkEnabled(m_bIsEnabled, enabled->toBool());`

uklotzde commented on src/widget/wspinny.cpp:
```diff
@@ -526,9 +526,9 @@ void WSpinny::updateVinylControlSignalEnabled(double enabled) {
         return;
     }
-    m_bSignalActive = enabled;
+    m_bSignalActive = (enabled != 0);
```
redundant parentheses

uklotzde commented on src/widget/wspinny.cpp:
```diff
@@ -539,7 +539,7 @@ void WSpinny::updateVinylControlSignalEnabled(double enabled) {
 
 void WSpinny::updateVinylControlEnabled(double enabled) {
-    m_bVinylActive = enabled;
+    m_bVinylActive = (enabled != 0);
```
redundant parentheses

uklotzde commented on src/widget/wtracktableview.cpp:
```diff
@@ -946,7 +946,7 @@ void WTrackTableView::applySorting() {
     }
 
-    Qt::SortOrder sortOrder = m_pSortOrder->get() ? Qt::DescendingOrder : Qt::AscendingOrder;
+    Qt::SortOrder sortOrder = (m_pSortOrder->get() == 0) ? Qt::AscendingOrder : Qt::DescendingOrder;
```
toBool()?

Holzhaus replied:
That would work, but IMHO it's semantically correct. The sort order is an enumeration with two values, not a boolean.

Holzhaus committed:
WSpinny: Remove redundant parentheses

Holzhaus committed:
VinylControlXwax: Simplify boolean expression

Holzhaus committed:
BalanceEffect: Use double for BalanceGroupState::m_freq

Holzhaus committed:
ControlObject: Simplify and fix static toBool() method

The comparison was changed from `value != 0` to `value > 0`. This is
unintuitive and misleading, but it needs to be consistent with the
non-static toBool() implementation which does the same. A different
outcome depending on whether you call the static or non-static version
would be even more confusing.

daschuer commented on src/engine/controls/bpmcontrol.cpp:
```diff
@@ -369,14 +378,14 @@ double BpmControl::calcSyncedRate(double userTweak) {
     // or we're in reverse, just return the rate as-is.
-    if (!m_pQuantize->get() || isMaster(getSyncMode()) ||
-            !m_pBeats || m_pReverseButton->get()) {
+    if (m_pQuantize->toBool() || isMaster(getSyncMode()) ||
```
Missing !

Holzhaus replied:
good catch. Looks like we don't have enough tests... :disappointed:

daschuer commented on src/engine/controls/cuecontrol.cpp:
```diff
@@ -853,9 +857,10 @@ void CueControl::hotcueActivatePreview(HotcueControl* pControl, double v) {
-void CueControl::hotcueClear(HotcueControl* pControl, double v) {
-    if (!v)
+void CueControl::hotcueClear(HotcueControl* pControl, double value) {
+    if (value != 0) {
```
Value == 0

daschuer commented on src/library/librarycontrol.cpp:
```diff
@@ -655,8 +655,8 @@ void LibraryControl::slotSortColumn(double v) {
-    if (static_cast<int>(m_pSortColumn->get()) == column) {
-        m_pSortOrder->set(!m_pSortOrder->get());
+    if (column == static_cast<int>(m_pSortColumn->get())) {
+        m_pSortOrder->set((m_pSortOrder->get() == 0) ? 1 : 0);
```
Now you set a double with an int instead of a boolean.
Was the compiler complaining? 
Is it satisfied now?
I think it should be 1.0 and 0.0.

Holzhaus replied:
No, the compiler doesn't complain. It didn't complain because we were setting a double with a bool btw, it complained because we were implicitly casting a double to bool with the `!` operator. Casting an int to bool doesn't lead to a precision loss, so I guess it's fine. I'll update the literals anyway.

daschuer reviewed (changes requested):
Some comments

Holzhaus committed:
EffectManifestParameter: Remove redundant parentheses

Holzhaus committed:
BpmControl: Fix wrong quantize check in calcSyncedRate()

This fixes a regression caused by c3cbb1d14186366f5d1d4c1100488f77df23c9c4.

Holzhaus committed:
CueControl: Fix wrong value guard in hotcueClear()

This fixes a regression introduced by 27ca8d7c85bb069e666b4f032d3735654aba4228.

Holzhaus committed:
LibraryControl: Clean up slotSortColumnToggle()

Holzhaus committed:
BalanceEffect: Remove wrong freq typecasts to int

Since commit 5b2c009520e1ee3bdd5c29bb477dd09000f51ad0 the `freq`
variable is a double, not an int.

Holzhaus commented on src/effects/builtin/balanceeffect.cpp:
```diff
@@ -128,7 +128,7 @@ void BalanceEffect::processChannel(const ChannelHandle& handle,
     CSAMPLE_GAIN midSideStart = pGroupState->m_oldMidSide + midSideDelta;
 
-    int freq = pGroupState->m_freq;
+    double freq = pGroupState->m_freq;
```
I removed them.

Holzhaus committed:
ControlObject: Inline the static toBool() method

Holzhaus commented:
Ready to review.

daschuer reviewed (approved):
Thank you. LGTM

Holzhaus commented:
@poelzi Now that this has been merged, can you build Mixxx with `-DWARNINGS_FATAL=ON`?
